### PR TITLE
Add ImGui input controls for plugins

### PR DESCRIPTION
**Pull request by chreden**
Add bindings for ImGui input functions. This will at least let plugins do some basic input instead of checkboxes and buttons.
Rearrange files for the imgui project to `inc` and `src` - will probably do this for the rest of the projects at some point too.
#1129

### CODE REVIEW

File: trview.lua.imgui/src/trview.lua.imgui.input.cpp [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                { "InputDouble", input_double },
+                { "InputFloat", input_float },
+                { "InputFloat2", input_float2 },
+                { "InputFloat3", input_float3 },
+                { "InputFloat4", input_float4 },
+                { "InputInt", input_int },
+                { "InputInt2", input_int2 },
+                { "InputInt3", input_int3 },
+                { "InputInt4", input_int4 },
+                // { "InputScalar", input_scalar },
```
Comment (lahm86) [inline trview.lua.imgui/src/trview.lua.imgui.input.cpp:223]: Just checking if you want to leave these in, I noticed other commented lines had been removed in this file.

File: doc/lua/imgui.md [hunk 2]
```diff
@@ -27,14 +37,182 @@ The ImGui library allows plugins to render UI elements.
...
+| NoPadInnerX |
+| ScrollX |
+| ScrollY |
+| SortMulti |
+| SortTristate |
+
+## InputTextFlags
+
+| Name | Notes |
+| ---- ||
```
Comment (lahm86) [inline doc/lua/imgui.md:171]: This table doesn't render - I think it needs a hyphen under Notes.

**Review by lahm86 — COMMENTED** [review]
LGTM, just a couple of formatting things.

File: doc/lua/imgui.md (hunk 2 above)
Comment (chreden) [reply doc/lua/imgui.md:171]: Ah thanks, VS is more forgiving

File: trview.lua.imgui/src/trview.lua.imgui.input.cpp (hunk 1 above)
Comment (chreden) [reply trview.lua.imgui/src/trview.lua.imgui.input.cpp:223]: Thought I'd leave them here as I should probably add them at some point, I just don't know how to do it at the moment.

**Review by lahm86 — APPROVED** [review]

**Pull request merged**